src/commands.ts

```diff
@@ -0,0 +1,168 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import * as vscode from 'vscode';
+import * as path from 'path';
+import * as Utils from './utils';
+import * as Helper from './commonHelper';
+import * as strings from './strings';
+import { DotNetInfo, requireDotNetSdk, runDotNetCommand, findProjectTemplate } from './dotnet';
+import { CommandObserver } from './CommandObserver';
+import * as Constants from './constants';
+
+export default function registerCommands(commandObserver: CommandObserver, packageInfo: Utils.IPackageInfo): vscode.Disposable[] {
+    let dotNetSdkVersion = packageInfo.requiredDotNetCoreSDK;
+    return [
+        vscode.commands.registerCommand('pgproj.build.all', async () => {
+            requireDotNetSdk(dotNetSdkVersion).then(
+                async dotnet => {
+                    await vscode.window.withProgress({
+                        location: vscode.ProgressLocation.Notification,
+                        title: Constants.buildProgressTitle,
+                        cancellable: true
+                    }, async (progress, token) => {
+                        await buildAllProjects(dotnet, commandObserver, token);
+                });
+            });
+        }),
+        vscode.commands.registerCommand('pgproj.build.current', async (args) => {
+            requireDotNetSdk(dotNetSdkVersion).then(
+                async dotnet => {
+                    await vscode.window.withProgress({
+                        location: vscode.ProgressLocation.Notification,
+                        title: Constants.buildProgressTitle,
+                        cancellable: true
+                    }, async (progress, token) => {
+                        await buildCurrentProject(args, dotnet, commandObserver, token);
+                    });
+                });
+        }),
+        vscode.commands.registerCommand('pgproj.add.new', async (args) => {
+            requireDotNetSdk(dotNetSdkVersion).then(
+                async dotnet => {
+                    await addNewPostgreSQLProject(args, dotnet, packageInfo.projectTemplateNugetId, commandObserver);
+            })
+        }),
+    ];
+}
+
+async function buildCurrentProject(args, dotNetSdk: DotNetInfo, commandObserver: CommandObserver, cancelToken: vscode.CancellationToken) {
+    let project = '';
+    if (!args) {
+        const activeEditor = vscode.window.activeTextEditor;
+        if (activeEditor !== undefined) {
+            project = activeEditor.document.uri.fsPath;
+        }
+    } else {
+        project = args.fsPath;
+    }
+
+    commandObserver.clear();
```

Comment:
We need to make sure they don't run two builds at the same time or we should support it through ```commandObserver``` somehow. Can we check if there is a build already running and show them an error, saying cancel the current build before starting a new one?

Reply:
Vscode itself doesn't have any way to do this for commands (we can fix this when they move tasks over). We can handle this ourselves though.

Reply:
What does ```.clear()``` do? Should we first check if build is running, before clearing the output?